### DISCUSSION TITLE
UI/UX suggestion

### DISCUSSION THREAD

**Discussion by suonnon** [huchenlei/sd-webui-openpose-editor#20] [Ideas]
1. Reset pose:
- A "reset pose" button would be great. The button will reset the edited pose to the pose that before click the Edit button

2. Close button or Esc to close the popup editor:
- After click the edit button, I I change my mind, I can't see the close button or how to escape the popup editor?

Update: I see the close button, but it on the top right screen and so difficult to notice that it's there.

**Comment by huchenlei** [comment]
1. You can reset pose by clicking the run preprocessor button again
2. Close button is more visible in dark mode. Will address the issue in light mode soon.

**Reply by huchenlei** [reply]
2. Addressed in https://github.com/Mikubill/sd-webui-controlnet/pull/1621.

**Comment by Woisek** [comment]
I like to add also a suggestion:
Would it be possible to make the window with the image and the pose bigger in size? Or at least 768x768.
Reason is, when zooming in, the small width of the portrait size makes it a bit cumbersome to edit. A bigger width would help a lot
.An awesome possibility would, if we could resize the whole edit window, but I guess this can't be done as it is a canvas. 😕

**Reply by huchenlei** [reply]
Currently the canvas is 1:1 with the dimension used in ControlNet. We can probably make it 1:2 so that it looks better? Or should we calculate a ratio to make it take as much screen space as possible?

**Reply by Woisek** [reply]
>We can probably make it 1:2 so that it looks better? 

That would definitely be an improvement.

>Or should we calculate a ratio to make it take as much screen space as possible?

If this would be possible, why not. Users with a small screen would surely appreciate this. 😊
Thanks for looking into this. 👍

**Reply by huchenlei** [reply]
Canvas made bigger in v0.1.5.

**Reply by Woisek** [reply]
Cool! Will look at it ASAP. 👍

**Comment by suonnon** [comment]
@huchenlei  I have some more suggestions, I hope these could help the extension easier to use:

1. Able to moving around the preview (the key should be hold "Right Mouse" or "Middle Mouse" to move around the preview image. Yes, I know the zoom-in mouse position is good but sometime we also want to move around the image

2. Some text should be in light color instead (screenshot below) for Dark Theme users
![Screenshot 2023-06-12 223341](https://github.com/huchenlei/sd-webui-openpose-editor/assets/133345496/1e01c0e5-c2b3-475c-b3a6-0e888ee48250)

3. Undo/Redo and hotkey Ctrl + Z / Ctrl + Shift + Z support

**Reply by huchenlei** [reply]
1. I am not sure whether this will be addressed by solution to previous comment.
2. Filed #21 to track
3. Filed #22 to track

**Reply by suonnon** [reply]
> 1. I am not sure whether this will be addressed by solution to previous comment.
> 2. Filed [[Feature Request] Add dark mode support #21](https://github.com/huchenlei/sd-webui-openpose-editor/issues/21) to track
> 3. Filed [[Feature Request] Add Redo/Undo feature #22](https://github.com/huchenlei/sd-webui-openpose-editor/issues/22) to track

Yeah, if the preview frame of the image is larger, this "movement" issue/request is not necessary I guess. Love your quick responses and fixes! Can't wait for the next updates!!!

**Reply by huchenlei** [reply]
Canvas made bigger in v0.1.5.

**Comment by Azazellz** [comment]
Why is it so hard to access pose editing?
Just think:
  --You have to open a tab with a controlnet.
  -- You have to enable openpose model. 
  -- You have to upload the image.
  -- Then you need to generate the pose.
  -- And only then can you edit it in your extension.
So many extra steps!
What if I don't have a pose picture? What if pose picture in the wrong resolution\ration, and I should edit it? What if I want to set it manually, from scratch?

So, my suggestion - move your interface to a separate UI tab. 
Well, or just make an extra button with your UI somewhere nearby so that there's a quick access to making poses manually from scratch or from a reference picture.

Although maybe I just haven't found it and there is such a button somewhere?

**Reply by Kenjugs** [reply]
I had the same thoughts. I thought it would be possible with a JSON file, as there is an option to download a JSON of the pose from this extension, but how useful is that exactly? You can't upload the JSON to this extension unless you:

1) Have a base image in which you can preprocess to then press Edit button; or
2) Open the extension URL explicitly in a separate tab. But then you can't send the pose to controlnet. In this case you would need to download the image first, then do it the way from case 1). So... again, what's the point of the JSON?

The exported JSON from this extension is not compatible with fkunn1326's openpose editor (https://github.com/fkunn1326/openpose-editor.git) UI tab either, so you can't even do this as a 2-step process. Controlnet does not have any way to import JSON file either from what I can tell. Am I missing something with how to use this extension's JSON export?

**Reply by huchenlei** [reply]
There will be a button added to the ControlNet extension's image upload panel that allow you directly upload the JSON file later. Filed https://github.com/Mikubill/sd-webui-controlnet/issues/2236 for tracking.

**Reply by huchenlei** [reply]
@Kenjugs Feature implemented in ControlNet v1.1.419.

**Comment by godpunisher** [comment]
Hi there, I'm using this wonderful extension on automatic1111.
Lots of power users love to use ComfyUI.
Please make it available on ComfyUI too.

**Reply by huchenlei** [reply]
In ComfyUI, the openpose detection nodes only generate an image instead of a pose in JSON format. In order to implement the webui workflow in ComfyUI, we probably need to have custom openpose detection nodes that output JSON.

The embedding part of the extension will be straightforward as it just need to be in an iframe.

I will put this task on my TODO list.

**Reply by SRagy** [reply]
That would be great - this extension is quite central to my workflow. Thanks for making it!

**Comment by ride5k** [comment]
would be nice to have a way to multiple select points via mouse drag and quickly delete/inactivate them.
